bot/cogs/allow_deny_lists.py

```diff
@@ -95,9 +95,21 @@ async def _delete_data(self, ctx: Context, allowed: bool, list_type: ValidAllowD
         """Remove an item from an allow or denylist."""
         item = None
         allow_type = "whitelist" if allowed else "blacklist"
+        id_converter = IDConverter()
 
-        log.trace(f"Trying to delete the {content} item from the {list_type} {allow_type}")
+        # If this is a server invite, we need to convert it.
+        if list_type == "GUILD_INVITE" and not id_converter._get_id_match(content):
```

Comment:
To be clear, `_get_id_match` is checked because unlike `_add_data`,  you also want to permit deleting directly by ID?

Reply:
That is correct. There's no reason we can't allow deletion by ID, and in my opinion, rejecting deletion by ID would be worse UX. When you list all the guild invites, you get a list of IDs. It's quite natural to then try to remove one of those IDs, and kind of a pain in the ass to have to figure out what the corresponding invite is.

Reply:
Thanks for confirming.